rflx/specification/parser.py

```diff
@@ -307,29 +332,72 @@ def create_message(
             target_node = Field(name) if name else FINAL
             structure.append(Link(source_node, target_node))
 
+        if component.first != expr.UNDEFINED:
+            for l in structure:
+                if l.target.identifier == component.name:
+                    if l.first == expr.UNDEFINED:
+                        l.first = component.first
+                    else:
+                        error.append(
+                            f'first aspect of field "{component.name}" conflicts with previous'
+                            " specification",
+                            Subsystem.MODEL,
+                            Severity.ERROR,
+                            component.first.location,
+                        )
+                        error.append(
+                            "previous specification of first",
+                            Subsystem.MODEL,
+                            Severity.INFO,
+                            l.first.location,
+                        )
+
+        if component.length != expr.UNDEFINED:
+            for l in structure:
+                if l.target.identifier == component.name:
+                    if l.length == expr.UNDEFINED:
+                        l.length = component.length
+                    else:
+                        error.append(
+                            f'length aspect of field "{component.name}" conflicts with previous'
+                            " specification",
+                            Subsystem.MODEL,
+                            Severity.ERROR,
+                            component.length.location,
+                        )
+                        error.append(
+                            "previous specification of length",
+                            Subsystem.MODEL,
+                            Severity.INFO,
+                            l.length.location,
+                        )
+
+        if component.condition != expr.TRUE:
+            for l in structure:
+                if l.target.identifier == component.name:
+                    l.condition = (
+                        expr.And(component.condition, l.condition, location=l.condition.location)
+                        if l.condition != expr.TRUE
+                        else component.condition
+                    )
```

Comment:
Would it be possible to refactor this to only loop once through structure?

Reply:
Yes, that makes sense. Changed in 91cbcda72b8a1835084703c9748e12cd2b44efad.